hack/go-lint.sh

```diff
@@ -28,6 +28,6 @@ else
   $DOCKER run --rm \
     --volume "${PWD}:/go/src/github.com/openshift/sippy${VOLUME_OPTION}" \
     --workdir /go/src/github.com/openshift/sippy \
-    docker.io/golangci/golangci-lint:v1.56.2 \
+    docker.io/golangci/golangci-lint:v1.57.0 \
```

Comment:
This lets make lint work locally as it now is compatible with our go mod version. However it picks up a couple new issues we're not addressing yet. I'll try to get this in a future PR.

Reply:
Which is what CI uses but only after merge

Reply:
This implies fixing the lint warnings in this PR, which is probably ok, I'll do that now.